python/ray/data/dataset.py

```diff
@@ -826,6 +826,7 @@ def random_shuffle(
         *,
         seed: Optional[int] = None,
         num_blocks: Optional[int] = None,
+        **ray_remote_args,
```

Comment:
I have seen two styles of specifying remote args in dataset public APIs: one is `**ray_remote_args` and the other is `ray_remote_args: Dict[str, Any] = None`.

Should we just pick one for consistency?

Reply:
For the Dataset class APIs, it's actually quite consistent and using **ray_remote_args, see APIs like map_batches().

Reply:
what about `write_*()`

Reply:
They are deviated from write_datasource() which is also **ray_remote_args.
I think there are two questions:
- Which one is preferred? Probably the one with type annotation IMO, but a stronger code style guide is needed (currently Google python code style is used in theory, but 1) it's not that specific on issues like this; 2) it's not even quite followed in practice).
- Should we make them consistent? Yes, but this can be achieved only with a stronger code style guide, because **kwargs is quite common way to write arguments for many people v.s. type annotated args are nice for many people, this won't converge.
In either case, I think we are not trying to fix them here now though.

Reply:
Yea, we don't need to change anything in this PR. Just want to point that out so team is aware of this (something we should decide before GA).